iceoryx_examples/icedelivery/README.md

```diff
@@ -157,82 +152,72 @@ captured with the signal handler and stops the loop.
-```
 
-For quality of service a `popo::SubscriberOptions` object is created and the `queueCapacity` is set. This parameter
-specifies how many samples the queue of the subscriber object can hold. If the queue would encounter an overflow,
-the oldest sample is released to create space for the newest one, which is then stored.
 ```cpp
-iox::popo::SubscriberOptions subscriberOptions;
-subscriberOptions.queueCapacity = 10U;
+constexpr char APP_NAME[] = "iox-cpp-subscriber-untyped";
+iox::runtime::PoshRuntime::initRuntime(APP_NAME);
 ```
 
 In the next step a subscriber object is created, matching exactly the `capro::ServiceDescription` that the publisher
-offered. Additionally, the previously created subscriber options are passed to the constructor. If no subscriber options
-are created, a default value will be used which sets the queueCapacity to the maximum value:
+offered:
+
 ```cpp
-iox::popo::UntypedSubscriber untypedSubscriber({"Radar", "FrontLeft", "Object"}, subscriberOptions);
+iox::popo::UntypedSubscriber untypedSubscriber({"Radar", "FrontLeft", "Object"});
 ```
 
 When using the default n:m communication philosophy, the `SubscriptionState` is immediately `SUBSCRIBED`.
 However, when restricting iceoryx to the 1:n communication philosophy before being in the state `SUBSCRIBED`, the state is change to `SUBSCRIBE_REQUESTED`.
 
-Again in a while-loop we do the following: First check for the `SubscriptionState`
+Again in a while-loop we do the following:
+
 ```cpp
 while (!killswitch)
 {
-    if (untypedSubscriber.getSubscriptionState() == iox::SubscribeState::SUBSCRIBED)
-    {
-
+    untypedSubscriber.take()
+        .and_then([](const void* chunk)
+        {
+            // ...
+        })
+        .or_else([](auto& result)
+        {
+            if (result != iox::popo::ChunkReceiveResult::NO_CHUNK_AVAILABLE)
+            {
+                std::cout << "Error receiving chunk." << std::endl;
+            }
+        });
 ```
 
 The `killswitch` will be used to stop the programm execution.
 
-Once the publisher has sent data, we can receive the data
-```cpp
-untypedSubscriber.take()
-    .and_then([](iox::cxx::optional<iox::popo::Sample<const void>>& sample)
-    {
-        // ...
-    })
-    .if_empty([]
-    {
-        // ...
-    })
-    .or_else([](iox::popo::ChunkReceiveResult error)
-    {
-        std::cout << "Error receiving chunk." << std::endl;
-    });
-```
-
-Well, that's a bit of a [lambda](https://en.wikipedia.org/wiki/Anonymous_function#C++_(since_C++11)) jungle. Let's
-translate it into a story again: "Take the data and then if this succeeds, work with the sample, if the sample is empty
-do something different, or else if an error occured, print the string 'Error receiving chunk.'" Of course you don't
-need to take care about all cases, but it is advised to do so.
+Let's translate it into a story again: "Take the data and then if this succeeds, work with the sample, or else if an
```

Comment:
I love your stories! I think they will help a lot of people. 

Maybe you should place them even more prominently and introduce `and_then` / `or_else` before jumping into this code here with your splendid story!

Reply:
Thanks! I would add this to the new `expected` / `optional` document that @FerdinandSpitzschnueffler created.